vendor/brave-ios/Ads/BATBraveAds.mm

```diff
@@ -1016,6 +1026,21 @@ - (void)closeNotification:(const std::string &)id
   [self.notificationsHandler clearNotificationWithIdentifier:bridgedId];
 }
 
+- (void)recordAdEvent:(const std::string&)ad_type
+     confirmationType:(const std::string&)confirmation_type
+            timestamp:(const uint64_t)timestamp {
+  if (adEventHistory) {
+    adEventHistory->Record(ad_type, confirmation_type, timestamp);
+  }
+}
+
+- (std::vector<uint64_t>)getAdEvents:(const std::string&)ad_type
+                    confirmationType:(const std::string&)confirmation_type {
+  if (adEventHistory) {
+    return adEventHistory->Get(ad_type, confirmation_type);
+  }
```

Comment:
Won't compile, missing return statement for nil `adEventHistory`:

```suggestion
  if (adEventHistory) {
    return adEventHistory->Get(ad_type, confirmation_type);
  }
  return {};
```

Reply:
Fixed